pkg/didcomm/protocol/middleware/presentproof/middlewares.go

```diff
@@ -103,11 +114,73 @@ type presentationExchangePayload struct {
 	PresentationDefinition *presexch.PresentationDefinition `json:"presentation_definition"`
 }
 
+// OptPD represents option function for the PresentationDefinition middleware.
+type OptPD func(o *pdOptions)
+
+// WithAddProofFn allows providing function that will sign the Presentation.
+func WithAddProofFn(sign func(presentation *verifiable.Presentation) error) OptPD {
+	return func(o *pdOptions) {
+		o.addProof = sign
+	}
+}
+
+type pdOptions struct {
+	addProof func(presentation *verifiable.Presentation) error
+}
+
+func defaultPdOptions() *pdOptions {
+	return &pdOptions{
+		addProof: func(presentation *verifiable.Presentation) error {
+			return nil
+		},
+	}
+}
+
+// AddBBSProofFn add BBS+ proof to the Presentation.
+func AddBBSProofFn(p Provider) func(presentation *verifiable.Presentation) error {
+	const bls12381g2pub = 0xeb
+
+	km, cr := p.KMS(), p.Crypto()
+
+	return func(presentation *verifiable.Presentation) error {
+		bbsLoader, err := bbsJSONLDDocumentLoader()
+		if err != nil {
+			return err
+		}
+
+		kid, pubKey, err := km.CreateAndExportPubKeyBytes(kms.BLS12381G2Type)
+		if err != nil {
+			return err
+		}
+
+		methodID := fingerprint.KeyFingerprint(bls12381g2pub, pubKey)
+		didKey := fmt.Sprintf("%s#%s", fmt.Sprintf("did:key:%s", methodID), methodID)
+
+		presentation.Context = append(presentation.Context, bbsContext)
+
+		return presentation.AddLinkedDataProof(&verifiable.LinkedDataProofContext{
+			SignatureType:           "BbsBlsSignature2020",
+			SignatureRepresentation: verifiable.SignatureProofValue,
+			Suite: bbsblssignature2020.New(
+				suite.WithSigner(newBBSSigner(km, cr, kid)),
+				suite.WithVerifier(bbsblssignature2020.NewG2PublicKeyVerifier()),
```

Comment:
nit - you don't need a verifier when signing

Reply:
I don't think so. I will try to remove it.

Reply:
Done https://github.com/hyperledger/aries-framework-go/pull/2601